diff -u -r gridftp-hdfs-0.5.4.orig/configure.ac gridftp-hdfs-0.5.4/configure.ac
--- gridftp-hdfs-0.5.4.orig/configure.ac	2012-06-15 09:01:47.000000000 -0400
+++ gridftp-hdfs-0.5.4/configure.ac	2013-07-17 10:43:41.686071731 -0400
@@ -19,8 +19,8 @@
 AS_IF([ test -e ${JAVA_HOME}/include ], [JNIHDIR=${JAVA_HOME}/include], )
 AC_CHECK_FILE(${JNIHDIR}/jni.h, ,[AC_MSG_ERROR(Can't find jni.h in ${JNIHDIR})])
 
-AS_IF([ test -e ${JAVA_HOME}/jre/lib/amd64 ], LDFLAGS=["$LDFLAGS -L${JAVA_HOME}/jre/lib/amd64 -L${JAVA_HOME}/jre/lib/amd64/server"])
-AS_IF([ test -e ${JAVA_HOME}/jre/lib/i386 ], LDFLAGS=["$LDFLAGS -L${JAVA_HOME}/jre/lib/i386 -L${JAVA_HOME}/jre/lib/i386/server"])
+AS_IF([ test -e ${JAVA_HOME}/jre/lib/amd64 ], LDFLAGS=["$LDFLAGS -L${JAVA_HOME}/jre/lib/amd64 -L${JAVA_HOME}/jre/lib/amd64/server -Wl,-rpath,${JAVA_HOME}/jre/lib/amd64/server"])
+AS_IF([ test -e ${JAVA_HOME}/jre/lib/i386 ], LDFLAGS=["$LDFLAGS -L${JAVA_HOME}/jre/lib/i386 -L${JAVA_HOME}/jre/lib/i386/server -Wl,-rpath,${JAVA_HOME}/jre/lib/i386/server"])
 
 AC_ARG_WITH(hadoop, [  --with-hadoop=HADOOP_HOME Location of $HADOOP_HOME], [HADOOPHOME=${withval}], [HADOOPHOME=$HADOOP_HOME])
 AC_CHECK_FILE(${HADOOPHOME}/src/c++/libhdfs/hdfs.h, INCLUDE=["$INCLUDE -I${HADOOPHOME}/src/c++/libhdfs"],)
@@ -37,7 +37,8 @@
 AC_CHECK_FILE(${GLOBUSLOCATION}/include/gcc32, INCLUDE=["$INCLUDE -I${GLOBUSLOCATION}/include/gcc32"])
 
 # Checks for libraries.
-AC_CHECK_LIB([hdfs], [hdfsRead], , [AC_MSG_ERROR(Could not find libhdfs)], [-ljvm])
+AC_CHECK_LIB([jvm], [JNI_CreateJavaVM], , [AC_MSG_ERROR(Could not find libjvm)])
+AC_CHECK_LIB([hdfs], [hdfsRead], , [AC_MSG_ERROR(Could not find libhdfs)])
 #AC_CHECK_LIB([jsig], [signal], ,   [AC_MSG_ERROR(Could not find libjsig)], )
 
 # Checks for header files.
